### ISSUE TITLE
Full theme switcher

### ISSUE THREAD

**Issue by JulianCataldo** [JulianCataldo/astro#1]
- default theme (light)
- default theme (dark)
-

**Comment by ollebergkvist** [comment]
@JulianCataldo 

Any chance of adding tailwind support? :)

**Comment by JulianCataldo** [comment]
Hello @ollebergkvist !

I don't use Tailwind myself, as a "bare" SCSS fan.
Motivation of this project is future-proofing generic, low-levels, web patterns, with minimal dependencies.
I'm pretty sure SCSS will still be widely used in 10 years, while Tailwind might probably get 2 or 3 contenders in the meanwhile, it's not the same pace (and that's good!).
Also, more personally, I feel constrained by Tailwind, even if I understand the appeal of an higher level styles macros.

Still, people could build whatever they want as a design system on top of these _headless_ components.
But I'm totally open to facilitate its usage for, say, the color mode switcher (?).
Tell me if you want some API to be implemented for seamless Tailwind usage, and we will see.

**TL;DR**: Most of my **in-page** components are extending the generic Astro `HTMLElement`, so you should get auto-completion with `class` or `class:list` array, and pass your own Tailwind styles.

Cheers
